backend/LexBoxApi/Services/EmailService.cs

```diff
@@ -117,6 +119,17 @@ private async Task SendEmailAsync(MimeMessage message)
             throw;
         }
     }
+     private async Task SendEmailWithRetriesAsync(MimeMessage message, int retryCount = 3, int retryWaitSeconds = 5 * 60)
```

Comment:
5 minutes was pretty arbitrary, but I feel like 30 seconds is a bit too little for the general case. It's a parameter, so in the ForgotPassword case we can bump the retry up to 30 seconds (and increase the retry count) while leaving it at 5 minutes by default. Why 5 minutes? Because intuitively, that's how long I would expect a complicated bit of Kubernetes infrastructure, which was temporarily down and thus giving SMTP errors, to finish rebooting all its bits and pieces and become available. Just a gut feeling.

Reply:
OK. Yeah, then maybe just make the more "urgent" emails retry more and faster 🤷.

Reply:
I think we don't actually know what a sensible default is.  I could see how 5 minutes or 30 seconds could both be sensible defaults in different scenarios.  I think it's fine to just go with one and see if that works, as well as provide a sensible parameter for the one.

If it turns out that we learn of an email that is never sent out because of network/infrastructure outage and our retry logic failed too quickly, then maybe 30 seconds is too short

if it turns out that a user reports that they couldn't reset their password or receive an important registration email in a timely way, then the 5 minutes is too long

Reply:
I've made ForgotPassword retry every 30 seconds, and left 5 minute defaults for everything else. The email will be tried once immediately no matter what, and most of them will get through on the first try, so it's probably not something we need to worry about too much. (I also upped the retry count on ForgotPassword, because we don't want to give up on that one too easily).